test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,423 @@
+        });
+
+    auto channelOperation2 = PushNotificationManager::CreateChannelAsync(remoteId2);
+
+    channelOperation2.Completed(
+        [&channelEvent2, &channelOperationResult2](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                result.Channel().Close();
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                channelOperationResult2 = result.ExtendedError();
+            }
+
+            SetEvent(channelEvent2.get());
+        });
+
+
+    auto channelOperation3 = PushNotificationManager::CreateChannelAsync(remoteId3);
+
+    channelOperation3.Completed(
+        [&channelEvent3, &channelOperationResult3](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                result.Channel().Close();
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                channelOperationResult3 = result.ExtendedError();
+            }
+
+            SetEvent(channelEvent3.get());
+        });
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent2.get(), 960000 /* milliseconds */))
+    {
+        channelOperation2.Cancel();
+        channelOperationResult2 = ERROR_TIMEOUT;
+    }
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent3.get(), 960000 /* milliseconds */))
+    {
+        channelOperation3.Cancel();
+        channelOperationResult3 = ERROR_TIMEOUT;
+    }
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent1.get(), 960000 /* milliseconds */))
+    {
+        channelOperation1.Cancel();
+    }
+
+    return ((channelOperationResult2 == WPN_E_OUTSTANDING_CHANNEL_REQUEST) && (channelOperationResult3 == WPN_E_OUTSTANDING_CHANNEL_REQUEST));
+}
+
+bool ActivatorTest()
+{
+    try
+    {
+        PushNotificationActivationInfo info(
+            PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator,
+            c_fakeComServerId);
+
+        auto token = PushNotificationManager::RegisterActivator(info);
+        if (!token.TaskRegistration())
+        {
+            return false;
```

Comment:
This is insidious: If the test crashes before we reach line 332 (e.g., out of memory, time limit exceeded), then the background task remains registered, and the test will perpetually fail until the app is uninstalled. It perpetually fails because `RegisterActivator` will see that the background task is registered, so no new task will be registered.

As noted earlier, if `RegisterActivator` is able to register the task but cannot register the class object, then the call fails even though the task is still registered. That leaves this test in a bad state, because it assumes that the background task is never registered. All runs of this test will now fail due to a problem that occurred *on a previous run* (that wasn't even the test's fault). Our customers will not enjoy debugging this.